FWCore/Integration/test/test_TestInterProcessProd_cfg.py

```diff
@@ -0,0 +1,44 @@
+import FWCore.ParameterSet.Config as cms
+
+
+class ThingExternalProcessProducer(cms.EDProducer):
+    def __init__(self, prod):
+        self.__dict__['_prod'] = prod
+        super(cms.EDProducer,self).__init__('TestInterProcessProd')
+    def __setattr__(self, name, value):
+        setattr(self._prod, name, value)
+    def __getattr__(self, name):
+        if name =='_prod':
+            return self.__dict__['_prod']
+        return getattr(self._prod, name)
+    def clone(self, **params):
+        returnValue = ThingExternalProcessProducerProducer.__new__(type(self))
+        returnValue.__init__(self._prod)
```

Comment:
Here the `self._prod` is passed by reference, right? In the `SwitchProducer` the `clone()` also clones the contained EDProducers/EDAliases. I wonder if this should behave consistently.